pennylane/taskify.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains functionality to enable task-based workflows with devices.
+"""
+from typing import Union
+
+import pennylane as qml
+from pennylane.devices import TaskQubit
+
+import dask.distributed as dist
+
+
+def taskify_dev(dev: qml.Device, return_future: bool = False, gen_report: Union[bool, str] = False):
+    """
+    Returns a proxy-qubit device with the device argument as the intiantiable backend.
+
+    >>> d_dev = qml.device("default.qubit", wires=["a","b",2])
+    >>> t_dev = qml.taskify_dev(dev)
+    >>> <TaskQubit device (wires=3, shots=None) at 0x7f66fcbbeee0>
+    """
+    return TaskQubit(
+        dev.wires,
+        shots=dev.shots,
+        analytic=None,
+        backend=dev.short_name,
+        gen_report=gen_report,
+        future=return_future,
+    )
+
+
+def taskify(func, futures=False):
+    """
+    Converts a callable function into a scheduled task on the running backend client.
+    """
```

Comment:
ah, so this function is mainly a wrapper around client submission?

Reply:
I wonder if it is worthwhile as opposed to having users set up the client directly?

Reply:
I think having a facade here makes sense, as we can keep the same interface for use of `concurrent.futures` later, and let the changes happen under the hood.